### ISSUE TITLE
Element missing from accessibility snapshot

### ISSUE THREAD

**Issue by mayur-waghela** [microsoft/playwright-mcp#514]
I have noticed (even with the latest version of playwright mcp server) few elements from the webpage doesnt become part of the accessibility snapshot.. Refer the below screenshot from the webpage , the DOM and the accessibility snapshot.

The questions is, does the Playwright MCP server expects the DOM to be in some format or to follow some rules so that the object make up to the accessibility snapshot?

![Image](https://github.com/user-attachments/assets/ef3293e7-293b-49c2-80b2-3462d13f3a7c)

![Image](https://github.com/user-attachments/assets/40208f3d-c572-465f-ad83-2b15e86f33d9)

I cannot share the application link because it not a public facing app. Let me know if you need anything.

**Comment by pavelfeldman** [comment]
In devtools context menu pick "copy outer html" on body node and attach it here. That way we can try to repro

**Comment by mayur-waghela** [comment]
Its as below : @pavelfeldman 

[outerBody.txt](https://github.com/user-attachments/files/20579561/outerBody.txt)

**Comment by pavelfeldman** [comment]
Is this your page or you are trying to interact with a 3rd party page? The page itself insists that the label "Terms of service" should not be used to describe the input field. It explicitly sets description to "String (Text Area)".

![Image](https://github.com/user-attachments/assets/acdaeca4-25eb-47d7-8836-d527bbc275b9)

If that is your page, fix it. If not, note that Playwright does not use aria snapshot for AI, it provides a bit more information in the snapshot, so hopefully LLM can make sense of which text area is where.

**Comment by pavelfeldman** [comment]
Closing as per above.
